tcintercom/app/views.py

```diff
@@ -1,173 +1,129 @@
         return 'Reply successfully posted'
     else:
         return 'Company has support'
 
 
-async def create_issue(part, tags):
-    data = {
-        'title': 'From IC: ' + ','.join(tags),
-        'body': '**Created from intercom**\n\n' + part['body'],
-        'labels': tags,
-    }
-    await github_request('/issues', data)
-
-
-async def snooze_conv_for_close(conv):
-    id = conv['id']
-    admin_id = conv['assignee']['id'] or conf.ic_bot_id
-    a_week_hence = (datetime.now() + timedelta(days=7)).timestamp()
-    await intercom_request(
-        f'/conversations/{id}/reply',
-        data={'message_type': 'snoozed', 'snoozed_until': a_week_hence, 'admin_id': admin_id},
-        method='POST',
-    )
-
-
-async def check_message_tags(item: dict):
-    tags = [t['name'] for t in item['tags_added']['tags']]
-    if any(t in ['New help article', 'Update help article'] for t in tags):
-        await create_issue(item['conversation_parts']['conversation_parts'][0], tags)
-        return 'Issue created with tags'
-    if 'Snooze then close' in tags:
-        await snooze_conv_for_close(item)
-        return 'Conversation snoozed while waiting for reply'
-
-
-async def close_conv(conv: dict):
-    admin_id = conv['assignee']['id'] or conf.ic_bot_id
-    data = {'message_type': 'close', 'admin_id': admin_id, 'type': 'admin', 'body': CLOSE_CONV_TEMPLATE}
-    await intercom_request(f'/conversations/{conv["id"]}/parts', method='POST', data=data)
-
-
-async def check_unsnoozed_conv(item: dict):
-    tags = [t['name'] for t in item['tags']['tags']]
-    if 'Snooze then close' in tags:
-        # The normal item doesn't have the request info
-        conv_details = await intercom_request(f'/conversations/{item["id"]}')
-        conv_stats = conv_details['statistics']
-        a_week_prev = (datetime.now() - timedelta(days=7)).timestamp()
-        dt = max(conv_stats['last_contact_reply_at'] or 0, conv_stats['last_admin_reply_at'] or 0)
-        logger.info('Checking conv %s with last reply %s against %s', item['id'], dt, a_week_prev)
-        if dt < a_week_prev:
-            await close_conv(item)
-            return 'Conversation closed because of inactivity'
-
-
-async def callback(request: Request):
+async def handle_intercom_callback(request: Request) -> JSONResponse:
+    """
+    Handles the callback from Intercom and decides what actions to take based on the topic.
+    """
+    await validate_ic_webhook_signature(request)
     try:
         data = json.loads(await request.body())
     except ValueError:
         return JSONResponse({'error': 'Invalid JSON'}, status_code=400)
-    item_data = data['data']['item']
-    topic = data.get('topic')
+    item_data = data.get('data', {}).get('item', {})
+    topic = data.get('topic', None)
     msg = 'No action required'
+    logfire.info('Intercom callback topic={topic}', topic=topic, data=data)
```

Comment:
Ah I remember now why I only did logfire for this one, it's because it also sends all the data from the webhook which is helpful for debugging issues and I don't want that to be getting posted in papertrail.